### PR TITLE
feat: add route for getting partial task by id

### LINKED ISSUE #6: feat(route): add route for getting a single task
Add route under task for getting a task by it's id.

### PR DESCRIPTION
**Pull request by shimoncohen**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

Related issues: #6 
Closes #6

### CODE REVIEW

File: src/partialTask/models/partialTaskManager.ts [hunk 1]
```diff
@@ -2,21 +2,54 @@ import { DeleteResult } from 'typeorm';
...
 
-  public async getPartialTasksByDiscrete(discrete: DiscreteTaskEntity, order: SearchOrder): Promise<IPartialTaskResponse[]> {
+  public async getPartialTask(params: IPartialTaskParams): Promise<IPartialTaskResponse> {
     const repository = await this.getRepository();
-    const records = await repository.getAll(discrete, order);
 
+    const exists = await repository.exists(params);
+    // Check if discrete already exists
+    if (!exists) {
+      return Promise.reject();
```
Comment (cl-shaharshavit-rnd) [inline src/partialTask/models/partialTaskManager.ts:29]: add TODO for custom errors

File: src/partialTask/models/partialTaskManager.ts [hunk 2]
```diff
@@ -2,21 +2,54 @@ import { DeleteResult } from 'typeorm';
...
+    }
+
+    const record = await repository.get(params);
+    if (!record) {
+      return Promise.reject();
```
Comment (cl-shaharshavit-rnd) [inline src/partialTask/models/partialTaskManager.ts:34]: add TODO for custom errors

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]
Where are the tests???

**Comment by shimoncohen** [comment]
> Where are the tests???

Nowhere

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: tests/integration/discreteTask/discreteTask.spec.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+import {
+  discreteTaskGetError,
+  discreteTaskCreateOk,
+  discreteTaskDelete,
+  discreteTaskGetAll,
+  discreteTaskGetOk,
+  discreteTaskUpdateCompleted,
+  discreteTaskPutError,
+  discreteTaskPostError,
+  discreteTaskCreateError,
```
Comment (asafMasa) [inline tests/integration/discreteTask/discreteTask.spec.ts:25]: mind lint error

File: tests/integration/discreteTask/discreteTask.spec.ts [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    });
+
+    it('should return status code 400 on POST request with no body', async function () {
+      const discreteFindOneMock = discreteTaskRepositoryMocks.findOneMock;
+      const response = await requestSender.createResourceNoBody(discreteTaskPostError.params.id, discreteTaskPostError.params.version);
+      expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+      expect(discreteFindOneMock).toHaveBeenCalledTimes(0);
+    });
+
+    //   // All requests with status code of 400
```
Comment (asafMasa) [inline tests/integration/discreteTask/discreteTask.spec.ts:177]: why is this commented?

File: tests/integration/discreteTask/discreteTask.spec.ts [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+        ...discreteTaskPostError.params,
+        metadata: discreteTaskPostError.body.metadata,
+      } as IDiscreteTaskCreate);
+      expect(response.status).toBe(httpStatusCodes.INTERNAL_SERVER_ERROR);
+      // TODO: replace when error handling is added
+      // expect(response.status).toBe(httpStatusCodes.CONFLICT);
+    });
+
+    //   // All requests with status code 4XX-5XX
+    //   it('get should return status code 500 on db error', async function () {
```
Comment (asafMasa) [inline tests/integration/discreteTask/discreteTask.spec.ts:264]: why is this commented?

File: tests/integration/discreteTask/discreteTask.spec.ts [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+    //     findOneMock.mockRejectedValue(new Error('test Db error')); //TODO: replace with custom db errors
+
+    //     const response = await requestSender.getStatus();
+
+    //     expect(findOneMock).toHaveBeenCalledTimes(1);
+    //     expect(response.status).toBe(httpStatusCodes.INTERNAL_SERVER_ERROR);
+    //   });
+
+    //   it('update should return status code 500 on db error', async function () {
```
Comment (asafMasa) [inline tests/integration/discreteTask/discreteTask.spec.ts:273]: why is this commented?

File: tests/integration/partialTask/helpers/data.ts [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+const updateDate: string = new Date('10-10-2020').toUTCString();
+const minZoom = 0;
+const maxZoom = 9;
+
+const taskZoomBounds = {
+  minZoom,
+  maxZoom,
+};
+
+// const tasks = [
```
Comment (asafMasa) [inline tests/integration/partialTask/helpers/data.ts:22]: remove comments

File: tests/integration/partialTask/helpers/data.ts [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  body: {
+    status: Status.COMPLETED,
+  },
+};
+
+export const partialTaskGetError = {
+  params: partialTaskParams,
+};
+
+// export const discreteTaskGetAll = {
```
Comment (asafMasa) [inline tests/integration/partialTask/helpers/data.ts:101]: remove comments

File: tests/mocks/DiscreteManagerMock.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { ConnectionManager } from 'typeorm';
+import { DiscreteTaskManager } from '../../src/discreteTask/models/discreteTaskManager';
+
+let discreteTaskManager: DiscreteTaskManager;
```
Comment (asafMasa) [inline tests/mocks/DiscreteManagerMock.ts:4]: mind the lint comments

File: tests/integration/discreteTask/discreteTask.spec.ts (hunk 3 above)
Comment (cl-shaharshavit-rnd) [inline tests/integration/discreteTask/discreteTask.spec.ts:25]: remove if not used

File: tests/integration/partialTask/helpers/data.ts (hunk 7 above)
Comment (cl-shaharshavit-rnd) [inline tests/integration/partialTask/helpers/data.ts:22]: remove commented data if it isnt being used

File: tests/mocks/DiscreteManagerMock.ts (hunk 9 above)
Comment (cl-shaharshavit-rnd) [inline tests/mocks/DiscreteManagerMock.ts:4]: it doesnt seams to be used or exported

File: tests/mocks/DiscreteManagerMock.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+const isConnectedMock = jest.fn();
+const initMock = jest.fn();
+const getLayerHistoryRepository = jest.fn();
+const connectionManagerMock = ({
```
Comment (cl-shaharshavit-rnd) [inline tests/mocks/DiscreteManagerMock.ts:9]: it doesnt seams to be used or exported

File: tests/unit/discreteTask/discreteTask.spec.ts [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+      expect(discreteGetMock).toHaveBeenCalledTimes(1);
+      expect(discreteGetMock).toHaveBeenCalledWith(discreteTaskGet.params);
+    });
+
+    it('Fails to get non existing discrete task', async function () {
+      const discreteGetMock = discreteTaskRepositoryMocks.get;
+      discreteGetMock.mockResolvedValue(undefined);
+
+      try {
+        expect(await discreteTaskManager.getDiscreteTask(discreteTaskGet.params)).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/discreteTask/discreteTask.spec.ts:99]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/discreteTask/discreteTask.spec.ts [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+      const discreteExistsMock = discreteTaskRepositoryMocks.exists;
+      discreteExistsMock.mockResolvedValue(false);
+
+      try {
+        expect(
+          await discreteTaskManager.updateDiscreteTask({
+            ...discreteTaskUpdateError.params,
+            ...discreteTaskUpdateError.body,
+          })
+        ).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/discreteTask/discreteTask.spec.ts:148]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/discreteTask/discreteTask.spec.ts [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+      expect(discreteDeleteMock).toHaveBeenCalledTimes(1);
+      expect(discreteDeleteMock).toHaveBeenCalledWith(discreteTaskDelete.params);
+    });
+
+    it('Fails to delete non existing discrete task', async function () {
+      const discreteGetMock = discreteTaskRepositoryMocks.get;
+      discreteGetMock.mockResolvedValue(undefined);
+
+      try {
+        expect(await discreteTaskManager.deleteDiscreteTask(discreteTaskDeleteError.params)).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/discreteTask/discreteTask.spec.ts:186]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/partialTask/helpers/data.ts [hunk 14]
```diff
@@ -0,0 +1,103 @@
+import { Status } from '../../../../src/common/constants';
+import { IPartialTaskCreate } from '../../../../src/common/interfaces';
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/partialTask/helpers/data.ts:2]: remove if not used

File: tests/unit/partialTask/partialTask.spec.ts [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+      expect(partialGetMock).toHaveBeenCalledTimes(1);
+      expect(partialGetMock).toHaveBeenCalledWith(partialTaskGet.params);
+    });
+
+    it('Fails to get non existing partial task', async function () {
+      const partialExistsMock = partialTaskRepositoryMocks.exists;
+      partialExistsMock.mockResolvedValue(false);
+
+      try {
+        expect(await partialTaskManager.getPartialTask(partialTaskGetError.params)).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/partialTask/partialTask.spec.ts:96]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/partialTask/partialTask.spec.ts [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+      expect(partialGetAllMock).toHaveBeenCalledTimes(1);
+      expect(partialGetAllMock).toHaveBeenCalledWith(partialTaskGetByDiscrete.params, SearchOrder.DESC);
+    });
+
+    it('Fails to get all partial tasks by non existing discrete', async function () {
+      const discreteExistsMock = discreteTaskRepositoryMocks.exists;
+      discreteExistsMock.mockResolvedValue(false);
+
+      try {
+        expect(await partialTaskManager.getPartialTasksByDiscrete(partialTaskGetByDiscrete.params, SearchOrder.DESC)).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/partialTask/partialTask.spec.ts:128]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/partialTask/partialTask.spec.ts [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+      const partialExistsMock = partialTaskRepositoryMocks.exists;
+      partialExistsMock.mockResolvedValue(false);
+
+      try {
+        expect(
+          await partialTaskManager.updatePartialTask({
+            ...partialTaskUpdateError.params,
+            ...partialTaskUpdateError.body,
+          })
+        ).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/partialTask/partialTask.spec.ts:165]: add "Error" type to the toThrow and TODO to replace with custom type

File: tests/unit/partialTask/partialTask.spec.ts [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+      expect(partialDeleteMock).toHaveBeenCalledTimes(1);
+      expect(partialDeleteMock).toHaveBeenCalledWith(partialTaskDelete.params);
+    });
+
+    it('Fails to delete non existing partial task', async function () {
+      const partialGetMock = partialTaskRepositoryMocks.get;
+      partialGetMock.mockResolvedValue(partialTaskDeleteError.response);
+
+      try {
+        expect(await partialTaskManager.deleteResource(partialTaskDeleteError.params)).toThrowError();
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/partialTask/partialTask.spec.ts:196]: add "Error" type to the toThrow and TODO to replace with custom type

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: tests/unit/discreteTask/discreteTask.spec.ts (hunk 11 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/discreteTask/discreteTask.spec.ts:100]: should be 
toThrowError(Error)

File: tests/unit/discreteTask/discreteTask.spec.ts (hunk 12 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/discreteTask/discreteTask.spec.ts:150]: should be
toThrowError(Error)

File: tests/unit/discreteTask/discreteTask.spec.ts (hunk 13 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/discreteTask/discreteTask.spec.ts:189]: should be
toThrowError(Error)

File: tests/unit/partialTask/partialTask.spec.ts (hunk 15 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/partialTask/partialTask.spec.ts:97]: should be
toThrowError(Error)

File: tests/unit/partialTask/partialTask.spec.ts (hunk 16 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/partialTask/partialTask.spec.ts:130]: should be
toThrowError(Error)

File: tests/unit/partialTask/partialTask.spec.ts (hunk 17 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/partialTask/partialTask.spec.ts:168]: should be
toThrowError(Error)

File: tests/unit/partialTask/partialTask.spec.ts (hunk 18 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/partialTask/partialTask.spec.ts:200]: should be
toThrowError(Error)

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

**Review by cl-shaharshavit-rnd — APPROVED** [review]

**Pull request merged**